include/signalrclient/signalr_value.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+namespace signalr
+{
+    class value
+    {
+    public:
+        enum type
+        {
+            MAP,
+            ARRAY,
+            STRING,
+            NUMBER,
+            NULL,
```

Comment:
NULL is weird to have in a enum as that is a common type for win32 apps. NULL_VALUE?

Reply:
Could we use an `enum class` to get scoping?

Reply:
We would also promote this to not be nested, so you'd write `signalr::type::NULL`

Reply:
Still feels like it could be confusing, none of the types above are real "types".

Reply:
What about `signalr::type::nullValue` or `signalr::type::nullType`

Reply:
Ok, changed them to something different